src/plugins/layout/shadowdom.ts

```diff
@@ -15,15 +15,16 @@ export class ShadowDom {
   private removedNodes = this.doc.createElement("div");
   private shadowDomRoot = this.doc.createElement("div");
   private classifyNodes = false;
+  private nodes: IShadowDomNode[] = [];
```

Comment:
>nodes [](start = 10, length = 5)

nit: rename to something like indexToNodeMap? I tend to err on the side of too expressive, so maybe meet somewhere in the middle :P